### PR TITLE
WIP: SubsetSelect support for renaming subsets

### LINKED ISSUE #1539: Allow renaming of subsets
This ticket is to rename subsets to only alpha-numeric with spaces. This ticket should be wary about any backend dictionary caches that will also need to be updated.

[🐱](https://jira.stsci.edu/browse/JDAT-2350)

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request adds support in the SubsetSelect plugin component to handle renaming the selected subset via `subset_select.rename_selected('new name')`.  This would still need testing across all plugins, (further) consideration for https://github.com/spacetelescope/jdaviz/pull/1175#discussion_r829372470, and UI/UX for entering the new name and exposing name conflicts, etc.

Fix #1539 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/template_mixin.py [hunk 1]
```diff
@@ -397,15 +402,24 @@ def __init__(self, plugin, items, selected, selected_has_subregions=None,
...
     @staticmethod
     def _subset_type(subset):
         if isinstance(subset.subset_state, RoiSubsetState):
-            # then this is a spatial subset, we want to ignore
+            # then this is a spatial subset,
             return 'spatial'
         else:
+            # TODO: more advanced logic.  This is counting static regions in imviz as spectral
+            # (which isn't an issue so long as those are isolated to imviz which will never use
+            # subset selects with spectral regions)
```
Comment (pllim) [inline jdaviz/core/template_mixin.py:422]: What if one day, Cubeviz decides to use Imviz viewer? 😬

File: jdaviz/core/template_mixin.py [hunk 2]
```diff
@@ -476,6 +498,26 @@ def selected_obj(self):
...
+    def rename_selected(self, new_name):
+        if new_name in self.labels:
+            raise ValueError(f"{new_name} is already a named subset")
+        if new_name in ['Entire Spectrum', 'Surrounding']:
+            # these are names used in various subset dropdowns for other meanings
+            raise ValueError(f"{new_name} is a reserved name")
+
+        subset_group = self.selected_subset_group
+        if subset_group is None:
+            raise TypeError("current selection is not a subset")
```
Comment (pllim) [inline jdaviz/core/template_mixin.py:516]: Nitpicky nitpick:
```suggestion
            raise TypeError("Current selection is not a subset")
```

File: jdaviz/core/template_mixin.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/core/template_mixin.py:422]: then we'll need the more advanced logic (I'd probably suggest modifying the if-statement to catch the static regions as spatial and then implementing additional filters to be able to exclude them for aperture photometry).  We'll also need to do that if any plugins in Imviz support static regions and want them included in the dropdowns.

**Comment by kecnry** [comment]
If we decide to revive this [:cat:](https://jira.stsci.edu/browse/JDAT-2350), we could use similar UI to that in [lcviz's ephemeris component](https://github.com/spacetelescope/lcviz/pull/19).

**Pull request closed without merging**